activerecord/lib/active_record/reflection.rb

```diff
@@ -32,8 +32,11 @@ def self.create(macro, name, scope, options, ar)
     end
 
     def self.add_reflection(ar, name, reflection)
-      ar.clear_reflections_cache
-      ar._reflections = ar._reflections.merge(name.to_s => reflection)
+      # Add reflection to the current class, and downstream it to its descendants
+      (ar.descendants << ar).each do |record|
```

Comment:
If anyone has a way of making `ar.descendants << ar` look prettier, I'm all for it. All we're doing here is creating an array with all of the classes we're going to be adding the new reflection to (`ar`'s descendants and `ar` itself), to try and stay DRY. ☔